### DISCUSSION TITLE
bash: /usr/local/bin/nextflow: Permission denied

### DISCUSSION THREAD

**Discussion by GKerdivel** [nextflow-io/nextflow#2956] [Q&A]
## Bug report 

A colleague installed nextflow on our server following the documentation provided in this github (using curl). He gave execution right to all but yet only him can launch it using `nextflow run scrip.nf`. All other users have to launch `sudo nextflow run scrip.nf` or get the message 

> bash: /usr/local/bin/nextflow: Permission denied

### Expected behavior and actual behavior

Expected behavior : All users should be able to launch nextflow without superuser rights
Actual behavior : Apart from the one user who installed nextflow on the server, all other users have to be sudoers to launch nextflow

### Steps to reproduce the problem

NA

### Program output 

> bash: /usr/local/bin/nextflow: Permission denied

### Environment 

* Nextflow version: 21.10.6
* Java version: opendjdk 11.0.15
* Operating system: Ubuntu 20.04.4 LTS
* Bash version: 5.0.17

### Additional context

I tried to change the rights (`chmod 777`), changing the owner to and/or group to root or to change the group to a group common to all users. I also tried to change the localization of the nextflow file. None of that changed anything.

**Comment by mribeirodantas** [comment]
Can you run the command below and share the output here, please?
```
stat `whereis nextflow`
```

This command will give us information about the Nextflow binary, such as permissions, file owner and file owner group. Maybe that will help us understand what's going on.

**Reply by lholivera** [reply]
Same problem. This permissions: Access: (0711/-rwx--x--x)  Uid: ( 1003/leonidax)   Gid: ( 1003/leonidax)

**Reply by mgurbano** [reply]
Same problem. My permissions are slightly different than the comment above: Access: (0777/lrwxrwxrwx) Uid: (0/root) Gid: (0/root). I have 777 access... So I'm very unsure as to why my permission is denied.

**Reply by mribeirodantas** [reply]
Can you please confirm that you have permission for the whole path? If you have permission to execute a file but do not have permission to access the path where the file is located, you'll get permission denied. See the example below:
```
➜  ~ a/b/example.sh
oi
➜  ~ chmod 000 a
➜  ~ a/b/example.sh
zsh: permission denied: a/b/example.sh
```

**Reply by rusalkaguy** [reply]
I get this problem when trying to run a docker container (which uses nextflow internally) using singularity (thus, no root), because I'm on a shared cluster. 

`singularity exec docker://bioraddbg/sequoia_analysis_toolkit /bin/bash /opt/biorad/src/entrypoint.sh --help`
gives 
`/bin/bash: /usr/local/bin/nextflow: Permission denied`

while 
`singularity exec docker://bioraddbg/sequoia_analysis_toolkit ls -lsd /usr/ /usr/local /usr/local/bin /usr/local/bin/nextflow`
shows path is accessible and nextflow should be executable.
` 0 drwxr-xr-x 10 root root   162 Oct 10  2019 /usr/`
` 0 drwxr-xr-x 10 root root   135 Oct 10  2019 /usr/local`
` 0 drwxr-xr-x  2 root root   112 Oct 22  2020 /usr/local/bin`
`16 -rwx--x--x  1 root root 15478 Oct 22  2020 /usr/local/bin/nextflow`

**Comment by gelio5** [comment]
Get this problem too.
Error was in reading permission.
Solved by `sudo chmod 715 /usr/local/bin/nextflow `

**Comment by bounlu** [comment]
I have this issue every time I upgrade Nextflow with `sudo nextflow self-update`. The permission of the executable file `/usr/local/bin/nextflow` becomes `644`, and I have to manually change it to `755`. I believe `755` must be the default. Is there any reason why not so?

**Reply by pditommaso** [reply]
Nextflow cannot change the file permission because it would require `sudo`

**Reply by bounlu** [reply]
Yes I use `sudo` to upgrade nextflow for all the users and it reverts back the permissions.
